From a6eb22832dba1969134c3facbaf5961f901422ea Mon Sep 17 00:00:00 2001
From: Zheng Zengkai <zhengzengkai@huawei.com>
Date: Mon, 15 Nov 2021 19:35:44 +0800
Subject: [PATCH] kabi: Introduce KABI_RESERVE macro for openEuler kernel
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: a6eb22832dba1969134c3facbaf5961f901422ea


hulk inclusion
category: feature
bugzilla: https://gitee.com/openeuler/kernel/issues/I4GUAB
CVE: NA

-------------------------------

Add generic macro for kabi padding reserve for openEuler kernel.
This Macro should used to reserve kabi padding for base data structs
before kabi freeze.

Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Reviewed-by: Yang Yingliang <yangyingliang@huawei.com>
Reviewed-by: Xie XiuQi <xiexiuqi@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 include/linux/kabi.h | 25 +++++++++++++++++++++++++
 1 file changed, 25 insertions(+)
 create mode 100644 include/linux/kabi.h

diff --git a/include/linux/kabi.h b/include/linux/kabi.h
new file mode 100644
index 000000000000..10c3dcfbbe55
--- /dev/null
+++ b/include/linux/kabi.h
@@ -0,0 +1,25 @@
+/* SPDX-License-Identifier: GPL-2.0 */
+/*
+ * kabi.h - openEuler kABI abstraction header
+ *
+ * Copyright (C) 2021. Huawei Technologies Co., Ltd. All rights reserved.
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 and
+ * only version 2 as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ */
+
+#ifndef _LINUX_KABI_H
+#define _LINUX_KABI_H
+
+/*
+ * Macro for Reserving KABI padding for base data structs before KABI freeze.
+ */
+
+#define KABI_RESERVE(n)		unsigned long kabi_reserved##n;
+
+#endif /* _LINUX_KABI_H */
-- 
2.26.2

